flink-runtime/src/test/java/org/apache/flink/runtime/io/network/netty/NettyMessageClientDecoderDelegateTest.java

```diff
@@ -0,0 +1,411 @@
+				BufferResponse expectedBufferResponse = (BufferResponse) expected[i];
+				BufferResponse decodedBufferResponse = (BufferResponse) decodedMessages.get(i);
+
+				assertEquals(expectedBufferResponse.backlog, decodedBufferResponse.backlog);
+				assertEquals(expectedBufferResponse.sequenceNumber, decodedBufferResponse.sequenceNumber);
+				assertEquals(expectedBufferResponse.bufferSize, decodedBufferResponse.bufferSize);
+				assertEquals(expectedBufferResponse.receiverId, decodedBufferResponse.receiverId);
+
+				if (expectedBufferResponse.bufferSize == 0 ||
+					expectedBufferResponse.receiverId.equals(context.getReleasedChannelId()) ||
+					expectedBufferResponse.receiverId.equals(context.getRemovedChannelId())) {
+
+					assertNull(decodedBufferResponse.getBuffer());
+				} else if (expectedBufferResponse.receiverId.equals(context.getNormalChannelId())) {
+					assertEquals(expectedBufferResponse.getBuffer(), decodedBufferResponse.getBuffer());
+				} else {
+					fail("The decoded buffer response was sent to unknown input channel.");
+				}
+			} else if (expected[i] instanceof NettyMessage.ErrorResponse) {
+				ErrorResponse expectedErrorResponse = (ErrorResponse) expected[i];
+				ErrorResponse decodedErrorResponse = (ErrorResponse) decodedMessages.get(i);
+
+				assertEquals(expectedErrorResponse.receiverId, decodedErrorResponse.receiverId);
+				assertEquals(expectedErrorResponse.cause.getClass(), decodedErrorResponse.cause.getClass());
+				assertEquals(expectedErrorResponse.cause.getMessage(), decodedErrorResponse.cause.getMessage());
+			} else {
+				fail("Unsupported message type");
+			}
+		}
+	}
+
+	private void releaseBuffers(ByteBuf[] buffers) {
+		for (ByteBuf buffer : buffers) {
+			buffer.release();
+		}
+	}
+
+	private Buffer createDataBuffer(int size) {
+		MemorySegment segment = MemorySegmentFactory.allocateUnpooledSegment(size);
+		NetworkBuffer buffer = new NetworkBuffer(segment, FreeingBufferRecycler.INSTANCE);
+		for (int i = 0; i < size / 4; ++i) {
+			buffer.writeInt(i);
+		}
+
+		return buffer;
+	}
+
+	private NettyChannelAndInputChannelIds createPartitionRequestClientHandler(int numberOfBuffersInNormalChannel) throws IOException, InterruptedException {
+		NetworkBufferPool networkBufferPool = new NetworkBufferPool(
+			numberOfBuffersInNormalChannel,
+			32 * 1024,
+			numberOfBuffersInNormalChannel);
+
+		SingleInputGate normalInputGate = createSingleInputGate(1);
+		RemoteInputChannel normalInputChannel = createRemoteInputChannel(
+			normalInputGate,
+			mock(PartitionRequestClient.class),
+			networkBufferPool);
+		normalInputGate.assignExclusiveSegments();
+		normalInputChannel.requestSubpartition(0);
+
+		SingleInputGate releasedInputGate = createSingleInputGate(1);
+		RemoteInputChannel releasedInputChannel = createRemoteInputChannel(
+			releasedInputGate,
+			mock(PartitionRequestClient.class),
+			networkBufferPool);
+		releasedInputGate.close();
```

Comment:
I suggest refactoring this method, because it is not necessary to create these channels for all the above tests.
Then we can further deduplicate the common codes:
                 ```
SingleInputGate normalInputGate = createSingleInputGate(1);
		RemoteInputChannel normalInputChannel = createRemoteInputChannel(
			normalInputGate,
			mock(PartitionRequestClient.class),
			networkBufferPool); 
```
And the class `NettyChannelAndInputChannelIds` is not necessary, we can create the separate required `InputChannel` for different test purpose.

Reply:
Fixed according to the comments.